### PR TITLE
feat(onboarding): Onboard with autodetected semanticCommits option

### LINKED ISSUE #2591: Configure an explicit value for 'semanticCommits' during onboarding
**What would you like Renovate to be able to do?**
For repositories that inconsistently use semantic commit messages style, the Renovate default of auto-detection can cause the issues seen in https://github.com/yarnpkg/yarn/pull/6463#issuecomment-426240070.

**Describe the solution you'd like**
When Renovate opens the onboarding PR, it could set either `semanticCommits: true/false` (or the equivalent presets) based on what auto-detection found at the time, so that the mode doesn't flip-flop back and forth in the future.

**Describe alternatives you've considered**
It might be that it's desirable for Renovate to automatically start using semantic commits mode, if a repository starts using that form permanently in the future - and so pinning in `renovate.json` not desired. If so, perhaps another way to avoid the issues seen in the yarn repository, would be for Renovate to add a warning comment to PRs saying "we notice that there are open Renovate PRs using both styles of commit messages. Please set `semanticCommits` explicitly to pick which style you'd like".

### PR DESCRIPTION
**Pull request by zharinov**
Closes #2591

Examples:
https://github.com/zharinov/diffparse/pull/1/files
https://github.com/zharinov/zod/pull/3

### CODE REVIEW

**Comment by zharinov** [comment]
I guess it can be used before #7170 merging as well

**Comment by zharinov** [comment]
Probability of wrong detection is pretty high, since we use shallow clone. But maybe it's not a big problem if we periodically do branch rebase.

**Comment by rarkins** [comment]
The base branch should be cloned for 10 commits.

**Comment by zharinov** [comment]
Can we increase this value? https://github.com/renovatebot/renovate/blob/master/lib/util/git/index.ts#L239

**Comment by rarkins** [comment]
I thought it was already 10. That line is a surprise to me.

**Comment by zharinov** [comment]
Need to check additionally check how can we clone with depth of `2`, but have the main branch with `10`

**Comment by viceice** [comment]
@zharinov we've discussed the git checkout and fetch done currently and will change bit a little bit.

Had not yet the time to add a issue for that.

Idea:

Do git init and configure on initRepo and only do a fetch on gitSync.


🤔

**Review by viceice — DISMISSED** [review]

File: lib/workers/repository/init/config.ts [hunk 1]
```diff
@@ -220,9 +220,9 @@ export async function getRepoConfig(
 ): Promise<RenovateConfig> {
   let config = { ...config_ };
   config.baseBranch = config.defaultBranch;
-  config = await checkOnboardingBranch(config);
-  config = await mergeRenovateConfig(config);
   config.semanticCommits =
     config.semanticCommits ?? (await detectSemanticCommits());
+  config = await checkOnboardingBranch(config);
+  config = await mergeRenovateConfig(config);
```
Comment (rarkins) [inline lib/workers/repository/init/config.ts:226]: This reordering means that semantic commit detection will be done *before* `renovate.json` reading. Is that intentional? Will the user's config override it?

File: lib/workers/repository/onboarding/branch/config.ts [hunk 2]
```diff
@@ -1,9 +1,25 @@
...
+    return typeof config.semanticCommits === 'boolean'
+      ? config.semanticCommits
+      : false;
+  }
+  return onboardingConfig.semanticCommits;
+}
+
 export function getOnboardingConfig(config: RenovateConfig): string {
   const onboardingConfig = clone(config.onboardingConfig);
+  onboardingConfig.semanticCommits = getSemanticCommitsValue(
```
Comment (rarkins) [inline lib/workers/repository/onboarding/branch/config.ts:19]: Let's only set the value if it's `true`. I plan to default it to `false` in the next major release.

**Review by rarkins — DISMISSED** [review]

**Comment by zharinov** [comment]
More thorough testing needed:
- [ ] *Happy path* for semantic/regular repo onboarding
- [ ] Deprecated (boolean) values are always migrated
- Check of `semanticCommits` being read from different sources:
    - [ ] Presets
    - [ ] `renovate.json` (main branch)
    - [ ] `renovate.json` (onboarding branch)

**Comment by zharinov** [comment]
Superseded by #8237

**Pull request closed without merging**